### PR TITLE
prevent error in log: /etc/ddhcpd.hook: line 6: /etc/ddhcpd.d/*: not found

### PR DESCRIPTION
**Pull request by rubo77**
without this, this message appears every call in the output of `logread`

### CODE REVIEW

File: ddhcpd/files/etc/ddhcpd.hook [hunk 1]
```diff
@@ -1,7 +1,8 @@
 #!/bin/sh
 
-for i in /etc/ddhcpd.d/*
-do
-  $i "$@"
-done
-
+if [ -z "$(ls /etc/ddhcpd.d)" ]; then
```
Comment (christf) [inline ddhcpd/files/etc/ddhcpd.hook]: do not parse the output of ls. Instead, please see here: https://stackoverflow.com/questions/4839214/does-ash-have-an-equivalent-to-bashs-nullglob-option

Comment (christf) [reply ddhcpd/files/etc/ddhcpd.hook]: the proper test would be for the executable flag.

**Comment by rubo77** [comment]
>  @christf:  do not parse the output of ls. Instead, please see here: https://stackoverflow.com/questions/4839214/does-ash-have-an-equivalent-to-bashs-nullglob-option
the proper test would be for the executable flag.

a check for executable would be of no use, because usually you **want** a warning if there is a file not executable.

I added a `-e` check now

File: ddhcpd/files/etc/ddhcpd.hook [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 for i in /etc/ddhcpd.d/*
 do
-  $i "$@"
+  if [ -e "$i" ] ; then
+    $i "$@"
```
Comment (BenBE) [inline ddhcpd/files/etc/ddhcpd.hook]: If quoting $i in the conditional you should be consistent and quote $i everywhere.

**Comment by rubo77** [comment]
added quotes

**Pull request merged**